common/src/main/java/com/wynntils/core/notifications/NotificationManager.java

```diff
@@ -29,25 +41,41 @@ public static MessageContainer queueMessage(TextRenderTask message) {
 
         WynntilsMod.info("Message Queued: " + message);
         MessageContainer msgContainer = new MessageContainer(message);
+        String messageText = message.getText();
 
-        WynntilsMod.postEvent(new NotificationEvent.Queue(msgContainer));
+        for (MessageContainer cachedContainer : cachedMessageSet) {
+            String checkableMessage = cachedContainer.getOriginalMessage();
+            if (messageText.equals(checkableMessage)) {
+                cachedContainer.incrementMessageCount();
 
-        // Overlay is not enabled, send in chat
-        if (!GameNotificationOverlayFeature.INSTANCE.isEnabled()) {
-            sendOrEditNotification(msgContainer);
+                doSendMessage(cachedContainer, true);
+                return cachedContainer;
+            }
         }
 
+        cachedMessageSet.put(msgContainer);
+
+        doSendMessage(msgContainer, false);
+
         return msgContainer;
     }
 
     public static void editMessage(MessageContainer msgContainer, String newMessage) {
         msgContainer.editMessage(newMessage);
 
-        WynntilsMod.postEvent(new NotificationEvent.Edit(msgContainer));
+        doSendMessage(msgContainer, true);
+    }
+
+    private static void doSendMessage(MessageContainer container, boolean isEdit) {
```

Comment:
Oh, drat, I didn't see it was different events. Having booleans in a method parameter list is seldom a good design choice. In this case, it is better to either instantiate and send in the correct Event from the call site, or move the postEvent stuff out of this function and just keep the check if overlay is not enabled. If you go the latter route, maybe rename it to `sendToChatIfNeeded` (??).

Reply:
I am sorry for suggesting a refactorization that did not really work out. I should have been more careful when looking at the code. :(